sale_general_discount/models/sale.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2018 Tecnativa - Sergio Teruel
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import api, fields, models
+from odoo.addons import decimal_precision as dp
+from lxml import etree
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    general_discount = fields.Float(
+        digits=dp.get_precision('Discount'),
+        string='Discount (%)',
+    )
+
+    @api.multi
+    @api.onchange('partner_id')
+    def onchange_partner_id(self):
+        for order in self:
+            super(SaleOrder, self).onchange_partner_id()
+            order.general_discount = order.partner_id.sale_discount
+
+    @api.multi
+    @api.onchange('general_discount')
+    def onchange_general_discount(self):
+        with self.env.norecompute():
+            for order in self:
+                order.mapped('order_line').update({
+                    'discount': order.general_discount,
+                })
+        self.recompute()
+
+    @api.model
+    def fields_view_get(self, view_id=None, view_type='form', toolbar=False,
+                        submenu=False):
+        res = super(SaleOrder, self).fields_view_get(
+            view_id, view_type, toolbar, submenu)
+        if view_type == 'form':
+            order_xml = etree.XML(res['arch'])
+            order_line_path = "//field[@name='order_line']"
```

Comment:
Can't we set the context [in the field directly](https://www.odoo.com/documentation/11.0/reference/orm.html#relational-fields)? Or can't we set a default method instead (in model `sale.order.line` and field `discount`; this default method could take into account `active_model` and `active_id` in the context)? :thinking:

Reply:
See my comment about that: https://github.com/OCA/sale-workflow/pull/685#discussion_r212824134

Reply:
Although your second proposal sounds good

Reply:
I discard the default value option, but i think that if active_... is in context so i will have the parent record to set de default value but I'm not sure.... I try it because the method fiels_view_get is wierd

Reply:
Have you tried the `add` argument in the XML?

Reply:
@Yajo I tested it, but in context has not active_model.. 
The new line that I am creating has not order_id record...